src/output.cpp

```diff
@@ -1990,6 +1990,46 @@ std::pair<std::string, nc_color> const& get_light_level(const float light)
     return strings[light_level];
 }
 
+template<typename RatingIterator>
+std::string get_labeled_bar( const double val, const int width, const std::string &label,
+    RatingIterator begin, RatingIterator end )
+{
+    std::string result;
+
+    result.reserve( width );
+    if( !label.empty() ) {
+        result += label;
+        result += ' ';
+    }
+    const int bar_width = width - utf8_width( result ) - 2; // - 2 for the brackets
+
+    result += '[';
+    if( bar_width > 0 ) {
+        int used_width = 0;
+        for( RatingIterator it(begin); it != end; ++it ) {
+            const double factor = std::min( 1.0, std::max( 0.0, it->first * val ) );
+            const int seg_width = int( factor * bar_width ) - used_width;
+
+            if( seg_width <= 0 ) {
+                continue;
+            }
+            used_width += seg_width;
+            result += std::string( seg_width, it->second );
```

Comment:
consider `std::string::insert` to avoid constructing an otherwise unused instance
eg. `result.insert( result.end(), seg_width, it->second )`